app/controllers/projects/queries_controller.rb

```diff
@@ -46,12 +46,34 @@ def new
 
   def create
     call = Queries::Projects::ProjectQueries::CreateService
-             .new(user: current_user)
+             .new(from: @query, user: current_user)
              .call(permitted_query_params)
```

Comment:
In other places of the application, we relied on `CopyServices` for doing operations like this. I think that would have the potential to keep the factory more focused of course at the expanse of a higher variety in the controller.

Reply:
I found usage of `CopyService`s only in api endpoints dedicated to copying, not in controllers. Also `BaseServices::Copy` is copying instance without changes. It is possible to first change an instance and then copy it, or even add overriding attributes in `…::ProjectQueries::CopyService`, but to me "creating from template" looks more straightforward.

Reply:
If I understand the need for doing all this correctly is that when you use "Save as" you might not have all the parameters sent along with the request but rather just an id of an existing query from which you want to deviate from. 

With the current approach I see the responsibility for all this split between the Controller (or the QueryLoading module), the factory and the create service. This is what I don't believe to be a good split especially since the parameter processing happens in two places. And the creation of the new query now also somewhat happens in the two places since the new query is instantiated in the factory and then passed to the CreateService to be overwritten and persisted. 

I would still think that having the responsibility of parameter processing in the service would be the cleanest solution but it seems you are right that most CopyServices don't take override parameters. Others do which makes it even weirder.

Would it be an option to place the parameters of the query one saves from (as identified by the query_id param) in the params passed into the create service? So that the whole parameters from the params sent with the request and also the ones taken from the source query are combined in `permitted_query_params` when they are passed to the create service?

Reply:
> Would it be an option to place the parameters of the query one saves from (as identified by the query_id param) in the params passed into the create service? So that the whole parameters from the params sent with the request and also the ones taken from the source query are combined in permitted_query_params when they are passed to the create service?

I've tried this at first, but didn't find a way to get attributes from the query in serialised form

Reply:
After looking into the code, I understand that struggle. 

Let's work with the current solution for now and create a code maintenance ticket to turn the copy services to take override parameters.